bundles/plugins/org.bonitasoft.studio.contract/src-test/java/org/bonitasoft/studio/contract/ui/wizard/FieldToContractInputMappingViewerCheckStateManagerTest.java

```diff
@@ -96,6 +97,54 @@ public void should_not_deselectParent_IfmappingHasNoParent() {
+        final FieldToContractInputMappingViewerCheckStateManager checkedStateManager = new FieldToContractInputMappingViewerCheckStateManager();
+
+        final FieldToContractInputMapping rootMapping1 = new RelationFieldToContractInputMapping(Mockito.mock(RelationField.class));
+        final SimpleFieldToContractInputMapping childMapping1 = new SimpleFieldToContractInputMapping(Mockito.mock(SimpleField.class));
+        final SimpleFieldToContractInputMapping childMapping2 = new SimpleFieldToContractInputMapping(Mockito.mock(SimpleField.class));
+        rootMapping1.addChild(childMapping1);
+        rootMapping1.addChild(childMapping2);
+        final SimpleFieldToContractInputMapping rootMapping2 = new SimpleFieldToContractInputMapping(Mockito.mock(SimpleField.class));
+        final ArrayList<FieldToContractInputMapping> newArrayList = Lists.newArrayList(rootMapping1, rootMapping2);
+
+        final CheckboxTreeViewer viewer = aCheckBoxTreeViewer(newArrayList);
+        checkedStateManager.checkAllStateChange(newArrayList, viewer, false);
+
+        assertThat(viewer.getChecked(rootMapping1)).isFalse();
+        assertThat(viewer.getChecked(rootMapping2)).isFalse();
+        assertThat(viewer.getChecked(childMapping1)).isFalse();
+        assertThat(viewer.getChecked(childMapping2)).isFalse();
+
+    }
+
+    @Test
+    public void should_select_all_when_checkedAllState_calledWithStateTrue() {
+        final FieldToContractInputMappingViewerCheckStateManager checkedStateManager = new FieldToContractInputMappingViewerCheckStateManager();
+
+        final FieldToContractInputMapping rootMapping1 = new RelationFieldToContractInputMapping(Mockito.mock(RelationField.class));
+        rootMapping1.setGenerated(false);
+        final SimpleFieldToContractInputMapping childMapping1 = new SimpleFieldToContractInputMapping(Mockito.mock(SimpleField.class));
+        final SimpleFieldToContractInputMapping childMapping2 = new SimpleFieldToContractInputMapping(Mockito.mock(SimpleField.class));
+        childMapping1.setGenerated(false);
+        childMapping2.setGenerated(false);
+        rootMapping1.addChild(childMapping1);
+        rootMapping1.addChild(childMapping2);
+        final SimpleFieldToContractInputMapping rootMapping2 = new SimpleFieldToContractInputMapping(Mockito.mock(SimpleField.class));
+        rootMapping2.setGenerated(false);
+        final ArrayList<FieldToContractInputMapping> newArrayList = Lists.newArrayList(rootMapping1, rootMapping2);
+
+        final CheckboxTreeViewer viewer = aCheckBoxTreeViewer(newArrayList);
+        checkedStateManager.checkAllStateChange(newArrayList, viewer, true);
+
+        assertThat(viewer.getChecked(rootMapping1)).isTrue();
+        assertThat(viewer.getChecked(rootMapping2)).isTrue();
+        assertThat(viewer.getChecked(childMapping1)).isTrue();
+        assertThat(viewer.getChecked(childMapping2)).isTrue();
+
+    }
+
```

Comment:
There is no test on a list partially selected.
Don't you think we should validate the behaviour of "select all" and "deselect all" when the user has already changed the current selection?